### PR TITLE
Cocotb tests update

### PR DESCRIPTION
**Pull request by M0stafaRady**
- Update cocotb tests to include more tests for gpios pull up and pull down and user configurations
- Update cocotb tests to work in gate level
- Optimize some tests and update the timeouts
- Update scripts and toplevel to include SDF files
- update path of sdf files to caravel/signoff/caravel/primetime-signoff

### CODE REVIEW

File: verilog/rtl/__user_project_wrapper.v [hunk 1]
```diff
@@ -108,42 +108,44 @@ wire [31:0] wbs_dat_o_gpio;
 wire [31:0] wbs_dat_o_debug;
 
 // reserve the last 2 regs for debugging registers
-`ifndef GPIO_TESTING
+// `ifndef GPIO_TESTING
```
Comment (marwaneltoukhy) [inline verilog/rtl/__user_project_wrapper.v:111]: Why did you remove these?

**Review by marwaneltoukhy — CHANGES_REQUESTED** [review]

File: verilog/rtl/__user_project_wrapper.v (hunk 1 above)
Comment (M0stafaRady) [reply verilog/rtl/__user_project_wrapper.v:111]: I needed 3 registers for testing user gpio so now module gpio_testing to use wbs_cyc_i_user instead of wbs_cyc_i_debug because the debug space can have up to 4 regs and 2 of them are already used so just for testing module gpio_testing will use user address space.
